src/libbson/src/bson/bson-decimal128.c

```diff
@@ -581,10 +581,12 @@ bson_decimal128_from_string_w_len (const char *string,     /* IN */
 #else
 #define SSCANF sscanf
 #endif
-      int read_exponent = SSCANF (++str_read, "%d%n", &exponent, &nread);
+      int read_exponent =
+         SSCANF (++str_read, "%" PRId64 "%n", &exponent, &nread);
```

Comment:
Please investigate the [test failures on Windows](https://parsley.mongodb.com/evergreen/mongo_c_driver_windows_2017_test_latest_server_ipv4_client_ipv4_noauth_nosasl_nossl_patch_797b371ce9c23a4a557cbd4ac5ac20ddd762c280_64b8476a3066154b1c1e58a9_23_07_19_20_28_28/0/task?bookmarks=0,6124&shareLine=930).

The definition of `SSCANF` differs on Windows and may be related to the error.

Reply:
It ended up being an issue with not converting the INT32 max and min values to `int64_t` :expressionless: